File: 05/Java/TripleExpression/Subtract.java
package TripleExpression;

public class Subtract extends BinaryOperator implements TripleExpression {
    public Subtract(TripleExpression first, TripleExpression second) {
        super(first, second);
    }

    protected double apply(double a, double b) {
        return a - b;
    }
}

File: 05/CSharp/TripleExpression_Classes_ONLY/Subtract.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TripleExpressionApp
{
    public class Subtract : BinaryOperator
    {
        public Subtract(ITripleExpression first, ITripleExpression second)
        {
            fOp = first;
            sOp = second;
        }

        override protected double Apply(double a, double b)
        {
            return a - b;
        }
    }
}
